tools: update capitalized-comments eslint rule

BridgeAR opened the pull request:
<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

Trott commented on .eslintrc.js:
```diff
@@ -60,9 +60,9 @@ module.exports = {
+        // Ignore all lines that have less characters than 20 and all lines that
         // start with something that looks like a variable name or code.
-        ignorePattern: '^.{0,30}$|^ [a-z]+ ?[0-9A-Z_.(/=:[#-]|^ std',
+        ignorePattern: '^.{0,20}$|^ [a-z]+ ?[0-9A-Z_.(/=:[#-]|^ std',
```
I don't think you need the `^` at the start or the `^ ` instances later in the regexp [because ESLint will only match on the start of the comment string excluding whitespace](https://github.com/eslint/eslint/blob/78358a8f66e95c4fcc921f2497e8a5ec5f1537ec/lib/rules/capitalized-comments.js#L94). (If you do take them out, test it of course....)

Trott commented on test/parallel/test-repl.js:
```diff
@@ -358,7 +358,7 @@ const errorTests = [
     expect: 'undefined'
   },
-  // npm prompt error message
+  // `npm` prompt error message.
```
I kind of chafe at markdown in comments, so this is (from my perspective) mildly unfortunate. Not blocking--only commenting in case you have another idea. (Maybe we can just add `|npm` to the end of the `ignorePatterns` regexp? Maybe long-term goal can be that the whole regexp will just be a set of words separated by pipe characters like `var|let|const|npm`?

BridgeAR replied:
We could add a couple of words to the blacklist but I guess it might become relatively big if we only use a blacklist. So I would definitely keep the current ignore pattern as it overall worked very well.

BridgeAR replied:
Hm, I personally would say this specific case actually benefits by making clear that `npm` is something specific. If it would be a more generic word, it would be difficult to distinguish it from the rest of the comment.

vsemozhetbyt commented on test/parallel/test-stream3-pause-then-read.js:
```diff
@@ -89,7 +89,7 @@ function onData() {
       if (seen > 100) {
-        // oh no, seen too much!
+        // Oh no, seen too much!
         // put the extra back.
```
```suggestion
        // Put the extra back.
```

BridgeAR commented:
Rebased due to conflicts. I incorporated all feedback in the "fixup" commits. PTAL.

Lite CI https://ci.nodejs.org/job/node-test-pull-request-lite-pipeline/3025/

BridgeAR commented:
Rebased due to conflicts.

Lite CI https://ci.nodejs.org/job/node-test-pull-request-lite-pipeline/3035/

BridgeAR commented:
Landed in b08a867d6016ccf04783a0f91fdbcc3460daf234 and 39141426d46a7e55d93ad2e8efa12ed86d223522.
